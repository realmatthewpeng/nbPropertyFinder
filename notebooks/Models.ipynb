{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# dependencies\n",
    "import pandas as pd\n",
    "import os\n",
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "import seaborn as sns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# File to Load \n",
    "file_to_load = \"County_Data.csv\"\n",
    "\n",
    "\n",
    "# Read Purchasing File and store into Pandas data frame\n",
    "county_data = pd.read_csv(file_to_load)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_county_data = pd.DataFrame(county_data)\n",
    "df_county_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Setup Axis\n",
    "state=df_county_data['State']\n",
    "county = df_county_data[\"County Name\"]\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "x_axis = np.arange(len(state))\n",
    "\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 45)\n",
    "plt.bar(x_axis, grad_rate, color=\"b\", align=\"center\")\n",
    "plt.title(\"County Graduation Rates\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.text(140, 0.6, \"Note:\\nGraduation Rates for all counties in NJ, NY, & PA.\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Graduation_Rates.png\", bbox_inches = \"tight\")\n",
    "# plt.zoom=10\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Povery"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Poverty Rate\"]\n",
    "x_axis = np.arange(len(pov_rate))\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 90)\n",
    "plt.bar(x_axis, pov_rate, color=\"r\", align=\"center\")\n",
    "plt.title(\"County Poverty Rates\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Poverty Rates\")\n",
    "plt.text(140, 30, \"Note:\\nPoverty Rates for all counties in NJ, NY, & PA.\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Poverty_Rates.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "pov_rate = df_county_data[\"Poverty Rate\"]\n",
    "\n",
    "plt.scatter(pov_rate, grad_rate)\n",
    "\n",
    "plt.grid()\n",
    "plt.title(\"High School Graduation Rates and Poverty Rates by County\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.xlabel(\"Poverty Rate\")\n",
    "# plt.text(50, 0.925, \"Note:\\nWe notice an inverse corellation ship between Graduation rates and Povetry rates.\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Grad_Poverty_Rates1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax1 = plt.subplots()\n",
    "\n",
    "tick_locations = [value for value in x_axis]\n",
    "\n",
    "plt.xticks(tick_locations, county, rotation=90)\n",
    "\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Poverty Rate\"]\n",
    "t = np.arange(len(county))\n",
    "ax1.plot(t, pov_rate, 'b-')\n",
    "ax1.set_xlabel('counties')\n",
    "# Make the y-axis label, ticks and tick labels match the line color.\n",
    "ax1.set_ylabel('Poverty Rate', color='b')\n",
    "ax1.tick_params('y', colors='b')\n",
    "\n",
    "plt.title(\"High School Graduation Rates and Poverty Rates by County\")\n",
    "\n",
    "ax2 = ax1.twinx()\n",
    "\n",
    "ax2.plot(t,grad_rate, 'r*')\n",
    "ax2.set_ylabel('Graduation Rate', color='r')\n",
    "ax2.tick_params('y', colors='r')\n",
    "zoom = 5\n",
    "w, h = fig.get_size_inches()\n",
    "fig.set_size_inches(w * zoom, h * zoom/2)\n",
    "# plt.xlim(0,20)\n",
    "# fig.tight_layout()\n",
    "plt.savefig(\"Images/County_Grad_Poverty_Rates2.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(8,6)) \n",
    "x = df_county_data[\"Poverty Rate\"] \n",
    "y = df_county_data[\"Graduation Rate\"] \n",
    "plt.scatter(x, y, color=\"g\", marker=\"o\", alpha=0.9) \n",
    "\n",
    "#Calculate and add R2 value\n",
    "mask = ~np.isnan(x) & ~np.isnan(y)\n",
    "\n",
    "\n",
    "#Add regression line\n",
    "sns.regplot(df_county_data[\"Poverty Rate\"], \n",
    "              df_county_data[\"Graduation Rate\"], color='r',label = \"Poverty \" )\n",
    "\n",
    "# Incorporate the other graph properties\n",
    "plt.title(\"Correlation between Graduation rates and Poverty rates\")\n",
    "plt.ylabel(\"Graduation rate\")\n",
    "plt.xlabel(\"Poverty rate\") \n",
    "\n",
    "plt.legend(loc='best')\n",
    "plt.grid(True)\n",
    "sns.set_style('whitegrid')\n",
    "plt.text(50, 0.925, \"Note:\\nWe notice an inverse corellation ship between Graduation rates and Povetry rates.\")\n",
    "\n",
    "plt.savefig(\"Images/County_Grad_Poverty_Rates3.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Household Size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "county = df_county_data[\"County Name\"]\n",
    "house_size = df_county_data[\"Household Size\"]\n",
    "\n",
    "x_axis = np.arange(len(house_size))\n",
    "\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 90)\n",
    "plt.bar(x_axis, house_size, color=\"r\", align=\"center\")\n",
    "plt.title(\"County Household Size\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Household Size\")\n",
    "plt.text(140, 2.5, \"Note:\\nHousehold size for all counties in NJ, NY, & PA.\")\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Household_Size.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "house_size = df_county_data[\"Household Size\"]\n",
    "\n",
    "plt.scatter(house_size, grad_rate)\n",
    "\n",
    "plt.grid()\n",
    "plt.title(\"High School Graduation Rates and Household Size by County\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.xlabel(\"Household Size\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Grad_House_Rates.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax1 = plt.subplots()\n",
    "\n",
    "tick_locations = [value for value in x_axis]\n",
    "\n",
    "plt.xticks(tick_locations, county, rotation= 90)\n",
    "\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Household Size\"]\n",
    "t = np.arange(len(county))\n",
    "ax1.plot(t, pov_rate, 'b-')\n",
    "ax1.set_xlabel('county')\n",
    "# Make the y-axis label, ticks and tick labels match the line color.\n",
    "ax1.set_ylabel('HouseHold Size', color='b')\n",
    "ax1.tick_params('y', colors='b')\n",
    "plt.title(\"High School Graduation Rates and Household Size by County\")\n",
    "ax2 = ax1.twinx()\n",
    "\n",
    "ax2.plot(t,grad_rate, 'r*')\n",
    "ax2.set_ylabel('Graduation Rate', color='r')\n",
    "ax2.tick_params('y', colors='r')\n",
    "zoom = 5\n",
    "w, h = fig.get_size_inches()\n",
    "fig.set_size_inches(w * zoom, h * zoom/2)\n",
    "fig.tight_layout()\n",
    "plt.savefig(\"Images/County_Grad_House_Rates1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(8,6)) \n",
    "x = df_county_data[\"Household Size\"]\n",
    "y = df_county_data[\"Graduation Rate\"] \n",
    "plt.scatter(x, y, color=\"g\", marker=\"o\", alpha=0.9) \n",
    "\n",
    "#Calculate and add R2 value\n",
    "mask = ~np.isnan(x) & ~np.isnan(y)\n",
    "\n",
    "\n",
    "#Add regression line\n",
    "sns.regplot(df_county_data[\"Household Size\"], \n",
    "              df_county_data[\"Graduation Rate\"], color='r',label = \"Household Size\" )\n",
    "\n",
    "# Incorporate the other graph properties\n",
    "plt.title(\"Correlation between Graduation rates and Household Size\")\n",
    "plt.ylabel(\"Graduation Rate\")\n",
    "plt.xlabel(\"House Size\") \n",
    "\n",
    "plt.legend(loc='best')\n",
    "plt.grid(True)\n",
    "sns.set_style('whitegrid')\n",
    "plt.text(3.4, 0.925, \"Note:\\nThere is a slight inverse correlation shown in this graph.\")\n",
    "\n",
    "plt.savefig(\"Images/County_Grad_House_Rates2.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Unemployment Rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "county = df_county_data[\"County Name\"]\n",
    "house_size = df_county_data[\"Unemployment Rate\"]\n",
    "x_axis = np.arange(len(house_size))\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 90)\n",
    "plt.bar(x_axis, house_size, color=\"r\", align=\"center\")\n",
    "plt.title(\"County Unemployment Rate\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Unemployment Rate\")\n",
    "plt.text(140, 7, \"Note:\\nUnemployment Rates for all counties in NJ, NY, & PA.\")\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Unemployment1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "house_size = df_county_data[\"Unemployment Rate\"]\n",
    "\n",
    "plt.scatter(house_size, grad_rate)\n",
    "\n",
    "plt.grid()\n",
    "plt.title(\"High School Graduation Rates and Unemployment Rate by County\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.xlabel(\"Unemployment Rate\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Grad_Unemployment_Rates.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax1 = plt.subplots()\n",
    "\n",
    "tick_locations = [value for value in x_axis]\n",
    "\n",
    "plt.xticks(tick_locations, county, rotation= 90)\n",
    "\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Unemployment Rate\"]\n",
    "t = np.arange(len(county))\n",
    "ax1.plot(t, pov_rate, 'b-')\n",
    "ax1.set_xlabel('county')\n",
    "# Make the y-axis label, ticks and tick labels match the line color.\n",
    "ax1.set_ylabel('Unemployment Rate', color='b')\n",
    "ax1.tick_params('y', colors='b')\n",
    "\n",
    "ax2 = ax1.twinx()\n",
    "plt.title(\"High School Graduation Rates and Unemployment Rate by County\")\n",
    "ax2.plot(t,grad_rate, 'r*')\n",
    "ax2.set_ylabel('Graduation Rate', color='r')\n",
    "ax2.tick_params('y', colors='r')\n",
    "zoom = 5\n",
    "w, h = fig.get_size_inches()\n",
    "fig.set_size_inches(w * zoom, h * zoom/2)\n",
    "fig.tight_layout()\n",
    "plt.savefig(\"Images/County_Grad_Unemployment Rate1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(8,6)) \n",
    "x = df_county_data[\"Unemployment Rate\"]\n",
    "y = df_county_data[\"Graduation Rate\"] \n",
    "plt.scatter(x, y, color=\"g\", marker=\"o\", alpha=0.9) \n",
    "\n",
    "#Calculate and add R2 value\n",
    "mask = ~np.isnan(x) & ~np.isnan(y)\n",
    "\n",
    "\n",
    "#Add regression line\n",
    "sns.regplot(df_county_data[\"Unemployment Rate\"], \n",
    "              df_county_data[\"Graduation Rate\"], color='r',label = \"Unemployment Rate\" )\n",
    "\n",
    "# Incorporate the other graph properties\n",
    "plt.title(\"Correlation between Graduation Rates and Unemployment Rate\")\n",
    "plt.ylabel(\"Graduation Rate\")\n",
    "plt.xlabel(\"Unemployment Rate\") \n",
    "\n",
    "plt.legend(loc='best')\n",
    "plt.grid(True)\n",
    "sns.set_style('whitegrid')\n",
    "plt.text(12, 0.925, \"Note:\\nAreas with higher unemployment tend to have a lower graduation rate.\")\n",
    "\n",
    "plt.savefig(\"Images/County_Grad_Unemployment Rate2.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Median Income"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "county = df_county_data[\"County Name\"]\n",
    "house_size = df_county_data[\"Median Income\"]\n",
    "x_axis = np.arange(len(house_size))\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 90)\n",
    "plt.bar(x_axis, house_size, color=\"r\", align=\"center\")\n",
    "plt.title(\"County Median Income\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Median Income Rate\")\n",
    "plt.text(140, 120000, \"Note:\\nMedian Income for all counties in NJ, NY, & PA.\")\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Median Income1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "house_size = df_county_data[\"Median Income\"]\n",
    "\n",
    "plt.scatter(house_size, grad_rate)\n",
    "\n",
    "plt.grid()\n",
    "plt.title(\"High School Graduation Rates and Median Income by County\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.xlabel(\"Median Income\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Grad_Median Income1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax1 = plt.subplots()\n",
    "\n",
    "tick_locations = [value for value in x_axis]\n",
    "\n",
    "plt.xticks(tick_locations, county, rotation= 90)\n",
    "\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Median Income\"]\n",
    "t = np.arange(len(county))\n",
    "ax1.plot(t, pov_rate, 'b-')\n",
    "ax1.set_xlabel('county')\n",
    "# Make the y-axis label, ticks and tick labels match the line color.\n",
    "ax1.set_ylabel('Median Income', color='b')\n",
    "ax1.tick_params('y', colors='b')\n",
    "plt.title(\"High School Graduation Rates and Median Income by County\")\n",
    "ax2 = ax1.twinx()\n",
    "\n",
    "ax2.plot(t,grad_rate, 'r*')\n",
    "ax2.set_ylabel('Graduation Rate', color='r')\n",
    "ax2.tick_params('y', colors='r')\n",
    "zoom = 5\n",
    "w, h = fig.get_size_inches()\n",
    "fig.set_size_inches(w * zoom, h * zoom/2)\n",
    "fig.tight_layout()\n",
    "plt.savefig(\"Images/County_Grad_Median Income2.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(8,6)) \n",
    "x = df_county_data[\"Median Income\"]\n",
    "y = df_county_data[\"Graduation Rate\"] \n",
    "plt.scatter(x, y, color=\"g\", marker=\"o\", alpha=0.9) \n",
    "\n",
    "#Calculate and add R2 value\n",
    "mask = ~np.isnan(x) & ~np.isnan(y)\n",
    "\n",
    "\n",
    "#Add regression line\n",
    "sns.regplot(df_county_data[\"Median Income\"], \n",
    "              df_county_data[\"Graduation Rate\"], color='r',label = \"Median Income\" )\n",
    "\n",
    "# Incorporate the other graph properties\n",
    "plt.title(\"Correlation between Graduation rates and Median Income\")\n",
    "plt.ylabel(\"Graduation rate\")\n",
    "plt.xlabel(\"Median Income\") \n",
    "\n",
    "plt.legend(loc='best')\n",
    "plt.grid(True)\n",
    "sns.set_style('whitegrid')\n",
    "plt.text(170000, 0.925, \"Note:\\nAreas with higher median income tend to have a higher graduation rate.\")\n",
    "\n",
    "plt.savefig(\"Images/County_Grad_Median Income3.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Speak a language other than English"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "county = df_county_data[\"County Name\"]\n",
    "house_size = df_county_data[\"Speak a language other than English\"]\n",
    "x_axis = np.arange(len(house_size))\n",
    "# Create a bar chart based upon the above data\n",
    "tick_locations = [value for value in x_axis]\n",
    "# plt.xticks(tick_locations, county, rotation= 90)\n",
    "plt.bar(x_axis, house_size, color=\"r\", align=\"center\")\n",
    "plt.title(\"County ESL\")\n",
    "plt.xlabel(\"Counties\")\n",
    "plt.ylabel(\"Speak a language other than English\")\n",
    "plt.text(140, 40, \"Note:\\nSpoken languages beside English for all counties in NJ, NY, & PA.\")\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Speak a language other than English.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "house_size = df_county_data[\"Speak a language other than English\"]\n",
    "\n",
    "plt.scatter(house_size, grad_rate)\n",
    "\n",
    "plt.grid()\n",
    "plt.title(\"High School Graduation Rates and ESL by County\")\n",
    "plt.ylabel(\"Graduation Rates\")\n",
    "plt.xlabel(\"Speak a language other than English\")\n",
    "\n",
    "# Save an image of the chart and print it to the screen\n",
    "plt.savefig(\"Images/County_Grad_Speak a language other than English1.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig, ax1 = plt.subplots()\n",
    "\n",
    "tick_locations = [value for value in x_axis]\n",
    "\n",
    "plt.xticks(tick_locations, county, rotation= 90)\n",
    "\n",
    "grad_rate = df_county_data[\"Graduation Rate\"]\n",
    "county = df_county_data[\"County Name\"]\n",
    "pov_rate = df_county_data[\"Speak a language other than English\"]\n",
    "t = np.arange(len(county))\n",
    "ax1.plot(t, pov_rate, 'b-')\n",
    "ax1.set_xlabel('county')\n",
    "# Make the y-axis label, ticks and tick labels match the line color.\n",
    "ax1.set_ylabel('Speak a language other than English', color='b')\n",
    "ax1.tick_params('y', colors='b')\n",
    "\n",
    "plt.title(\"High School Graduation Rates and ESL by County\")\n",
    "\n",
    "ax2 = ax1.twinx()\n",
    "\n",
    "ax2.plot(t,grad_rate, 'r*')\n",
    "ax2.set_ylabel('Graduation Rate', color='r')\n",
    "ax2.tick_params('y', colors='r')\n",
    "zoom = 5\n",
    "w, h = fig.get_size_inches()\n",
    "fig.set_size_inches(w * zoom, h * zoom/2)\n",
    "fig.tight_layout()\n",
    "plt.savefig(\"Images/County_Grad_Speak a language other than English2.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fig = plt.figure(figsize=(8,6)) \n",
    "x = df_county_data[\"Speak a language other than English\"]\n",
    "y = df_county_data[\"Graduation Rate\"] \n",
    "plt.scatter(x, y, color=\"g\", marker=\"o\", alpha=0.9) \n",
    "\n",
    "#Calculate and add R2 value\n",
    "mask = ~np.isnan(x) & ~np.isnan(y)\n",
    "\n",
    "\n",
    "#Add regression line\n",
    "sns.regplot(df_county_data[\"Speak a language other than English\"], \n",
    "              df_county_data[\"Graduation Rate\"], color='r',label = \"Speak a language other than English\" )\n",
    "\n",
    "# Incorporate the other graph properties\n",
    "plt.title(\"High School Graduation Rates and ESL by County\")\n",
    "plt.ylabel(\"Graduation Rate\")\n",
    "plt.xlabel(\"Speak a language other than English\") \n",
    "\n",
    "plt.legend(loc='best')\n",
    "plt.grid(True)\n",
    "sns.set_style('whitegrid')\n",
    "plt.text(65, 0.925, \"Note:\\nAreas with one or more Foreign languages beside English language \\ntend to have a Lower graduation rate.\")\n",
    "\n",
    "plt.savefig(\"Images/County_Grad_Speak a language other than English3.png\", bbox_inches = \"tight\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# for this one I am trying to put all factors we are comparing together but still I have to figure out how\n",
    "\n",
    "\n",
    "lines = df_county_data.plot.line()\n",
    "lines\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# if any of you feel like this is irrelevant plz feel free to delete it\n",
    "\n",
    "df_county_data.boxplot(\"Graduation Rate\", by=\"State\", figsize=(20, 10))\n",
    "\n",
    "plt.title(\"Compare Graduation Rates by States\")\n",
    "plt.xlabel(\"State\")\n",
    "plt.ylabel(\"Graduation Rate\")\n",
    "\n",
    "plt.savefig(\"Images/Graduation_State_Box_Plot.png\", bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Calculates correlations for stats we measured.\n",
    "\n",
    "corr_pov = abs(df_county_data[\"Poverty Rate\"].corr(grad_rate))\n",
    "corr_hh = abs(df_county_data[\"Household Size\"].corr(grad_rate))\n",
    "corr_job = abs(df_county_data[\"Unemployment Rate\"].corr(grad_rate))\n",
    "corr_inc = abs(df_county_data[\"Median Income\"].corr(grad_rate))\n",
    "corr_ESL = abs(df_county_data[\"Speak a language other than English\"].corr(grad_rate))\n",
    "\n",
    "print(\"Correlations\")\n",
    "print(\"Poverty:             \" + \"{:.4f}\".format(corr_pov))\n",
    "print(\"Median Income:       \" + \"{:.4f}\".format(corr_inc))\n",
    "print(\"Unemployment Rates:  \" + \"{:.4f}\".format(corr_job))\n",
    "print(\"Non-English at Home: \" + \"{:.4f}\".format(corr_ESL))\n",
    "print(\"Household Size:      \" + \"{:.4f}\".format(corr_hh))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "x_axis = [\"Poverty\", \"Med Inc\", \"Unemployment\", \"ESL\", \"HH Size\"]\n",
    "y_axis = [corr_pov, corr_inc, corr_job, corr_ESL, corr_hh]\n",
    "\n",
    "plt.bar(x_axis, y_axis, color=\"b\", align=\"center\")\n",
    "plt.title(\"Correlations of Demographic Metrics to High School Graduation Rates\")\n",
    "plt.xlabel(\"Demographic Metric\")\n",
    "plt.ylabel(\"Correlation\")\n",
    "\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
