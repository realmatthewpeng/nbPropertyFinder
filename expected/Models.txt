state=df_county_data['State']
county = df_county_data["County Name"]
grad_rate = df_county_data["Graduation Rate"]
x_axis = np.arange(len(state))
pov_rate = df_county_data["Poverty Rate"]
x_axis = np.arange(len(pov_rate))
t = np.arange(len(county))
x = df_county_data["Poverty Rate"] 
y = df_county_data["Graduation Rate"]
mask = ~np.isnan(x) & ~np.isnan(y)
house_size = df_county_data["Household Size"]
x_axis = np.arange(len(house_size))
pov_rate = df_county_data["Household Size"]
x = df_county_data["Household Size"]
house_size = df_county_data["Unemployment Rate"]
pov_rate = df_county_data["Unemployment Rate"]
x = df_county_data["Unemployment Rate"]
house_size = df_county_data["Median Income"]
pov_rate = df_county_data["Median Income"]
x = df_county_data["Median Income"]
house_size = df_county_data["Speak a language other than English"]